torch/csrc/api/src/nn/modules/transformer.cpp

```diff
@@ -422,21 +422,17 @@ Tensor TransformerImpl::generate_square_subsequent_mask(int64_t sz) {
   TORCH_CHECK(sz >= 0,
     "Input size must be non-negative to genearte a valid square subsequent mask, but got ", sz);
 
-  Tensor mask = (torch::triu(torch::ones({sz, sz})) == 1).transpose(0, 1).to(torch::kFloat32);
-
   // check IEEE754 support here since -inf is not guaranteed to be valid on non IEEE754 platform
   if (std::numeric_limits<float>::is_iec559) {
-    mask = mask.masked_fill(mask == 0, -std::numeric_limits<float>::infinity()).masked_fill(mask == 1, 0.f);
+    return torch::triu(torch::full({sz, sz}, -std::numeric_limits<float>::infinity()), 1);
```

Comment:
btw if this is a lot faster than expand(), should we file a bug? in theory expand-version allocates less and just needs a single value to be in cache